tecken/settings.py

```diff
@@ -6,765 +6,787 @@
-    # can be post-processed after.
-    # But we only need to write it down once per symbol. There's a memoizing
-    # guard and this defines how long it should cache that it memoized.
-    MEMOIZE_LOG_MISSING_SYMBOLS_SECONDS = values.IntegerValue(60 * 60 * 24)
-
-    # When we ask S3 for the size (if it exists) of a symbol already in S3
-    # this can be cached. This value determines how long we do that caching.
-    MEMOIZE_KEY_EXISTING_SIZE_SECONDS = values.IntegerValue(60 * 60 * 24)
-
-    # When we upload a .zip file, we iterate over the content and for each
-    # file within (that isn't immediately "ignorable") we kick off a
-    # function which figures out what (and how) to process the file.
-    # That function involves doing a S3 GET (technically ListObjectsV2),
-    # (possible) gzipping the payload and (possibly) a S3 PUT.
-    # All of these function calls get put in a
-    # concurrent.futures.ThreadPoolExecutor pool. This setting is about
-    # how many of these to start, max.
-    UPLOAD_FILE_UPLOAD_MAX_WORKERS = values.IntegerValue(default=None)
-
-    # Whether to store the missing symbols in Postgres or not.
-    # If you disable this, at the time of writing, missing symbols
-    # will be stored in the Redis default cache.
-    ENABLE_STORE_MISSING_SYMBOLS = values.BooleanValue(True)
-
-    # The prefix used when generating directories in the temp directory.
-    UPLOAD_TEMPDIR_PREFIX = values.Value("raw-uploads")
-
-    # When doing local development, especially load testing, it's sometimes
-    # useful to be able to bypass all URL checks for Upload by Download.
-    ALLOW_UPLOAD_BY_ANY_DOMAIN = values.BooleanValue(False)
-
-    # This is only really meant for the sake of being overrideable by
-    # other setting classes; in particular when running tests.
-    SYNCHRONOUS_UPLOAD_FILE_UPLOAD = values.BooleanValue(False)
-
-    DOWNLOAD_LEGACY_PRODUCTS_PREFIXES = [
-        "firefox",
-        "seamonkey",
-        "sunbird",
-        "thunderbird",
-        "xulrunner",
-        "fennec",
-        "b2g",
-    ]
-
-
-class Base(Core):
-    """Settings that may change per-environment, some with defaults."""
-
-    @classmethod
-    def setup(cls):
-        super().setup()
-
-        # For the sake of convenience we want to make UPLOAD_TRY_SYMBOLS_URL
-        # optional as an environment variable. If it's not set, set it
-        # by taking the UPLOAD_DEFAULT_URL and adding the prefix "/try"
-        # right after the bucket name.
-        if not cls.UPLOAD_TRY_SYMBOLS_URL:
-            default_url = urlparse(cls.UPLOAD_DEFAULT_URL)
-            path = default_url.path.split("/")
-            # Since it always start with '/', the point after the bucket
-            # name is the 3rd one.
-            path.insert(2, "try")
-            # Note `._replace` is actually not a private method.
-            try_url = default_url._replace(path="/".join(path))
-            cls.UPLOAD_TRY_SYMBOLS_URL = try_url.geturl()
```

Comment:
I took out computing this value. The comment suggests it's here for convenience, but what it really does is makes it difficult to move try symbols somewhere else. It should get set in the various environments. (Actually, all the "try" support should be redone.)